tests/cpp_integration_tests/test_main_model_se.cpp

```diff
@@ -293,22 +294,28 @@ TEST_CASE_TEMPLATE("Test main model - state estimation", CalculationMethod, Iter
         std::vector<AsymVoltageSensorUpdate> complete_asym_sensor_update{
             {4, 1.0, RealValue<asymmetric_t>{12.345e3}, RealValue<asymmetric_t>{0.1}}};
 
-        ConstDataset input_data;
-        input_data["node"] = ConstDataPointer{node_input.data(), static_cast<Idx>(node_input.size())};
-        input_data["source"] =
-            ConstDataPointer{incomplete_source_input.data(), static_cast<Idx>(incomplete_source_input.size())};
-        input_data["sym_voltage_sensor"] =
-            ConstDataPointer{incomplete_sym_sensor_input.data(), static_cast<Idx>(incomplete_sym_sensor_input.size())};
-        input_data["asym_voltage_sensor"] = ConstDataPointer{incomplete_asym_sensor_input.data(),
-                                                             static_cast<Idx>(incomplete_asym_sensor_input.size())};
-
-        ConstDataset update_data;
-        update_data["source"] =
-            ConstDataPointer{complete_source_update.data(), static_cast<Idx>(complete_source_update.size())};
-        update_data["sym_voltage_sensor"] =
-            ConstDataPointer{complete_sym_sensor_update.data(), static_cast<Idx>(complete_sym_sensor_update.size())};
-        update_data["asym_voltage_sensor"] =
-            ConstDataPointer{complete_asym_sensor_update.data(), static_cast<Idx>(complete_asym_sensor_update.size())};
+        ConstDataset input_data{false, 1, "input", meta_data::meta_data_gen::meta_data};
```

Comment:
here and in other places: do you think it's a good idea to use structured bindings?
```suggestion
        ConstDataset input_data{.is_batch = false, .batch_size = 1, "input", meta_data::meta_data_gen::meta_data};
```

Reply:
You can't. `Dataset` is not aggregate type.